### PR TITLE
test(rest-client): Add test for async rest client

### LINKED ISSUE #2119: NPE when using aync rest client
This will be fixed by https://github.com/resteasy/Resteasy/pull/1994, however we should add some tests

### PR DESCRIPTION
**Pull request by machi1990**
Fixes #2119. 

Also updates guide to mention support of `CompletionStage` as described in this comment: https://github.com/quarkusio/quarkus/issues/2119#issuecomment-484781468

/cc @stuartwdouglas @cescoffier

### CODE REVIEW

File: docs/src/main/asciidoc/rest-client-guide.adoc [hunk 1]
```diff
@@ -266,11 +268,101 @@ public class CountriesResourceTest {
 
 The code above uses link:http://rest-assured.io/[REST Assured]'s link:https://github.com/rest-assured/rest-assured/wiki/GettingStarted#jsonpath[json-path] capabilities.
 
+== Async Support
+The rest client supports asynchronous rest calls. Let's see it in action by adding a `getByNameAsync` method in our `CountriesService.class` rest interface. The code should look like
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client-guide.adoc]:
```suggestion
The rest client supports asynchronous rest calls. Let's see it in action by adding a `getByNameAsync` method in our `CountriesService` REST interface. The code should look like:
```

File: docs/src/main/asciidoc/rest-client-guide.adoc [hunk 2]
```diff
@@ -266,11 +268,101 @@ public class CountriesResourceTest {
...
+             "[0].currencies[0].name", is("Euro")
+        );
+    }
+----
+
 == Package and run the application
 
 Run the application with: `./mvnw compile quarkus:dev`.
 Open your browser to http://localhost:8080/country/name/greece.
-
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-client-guide.adoc]: Why that?

**Review by gsmet — DISMISSED** [review]
I added a few comments for the doc.

File: docs/src/main/asciidoc/rest-client-guide.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/rest-client-guide.adoc]: I wanted to mention the possibility to open the browser for the `name-async` endpoint. At the end I dropped the idea without properly formatting the paragraph. I'll revert.

**Comment by machi1990** [comment]
> I added a few comments for the doc.

@gsmet Thanks for the review. I have taken into account your feedbacks.

**Comment by machi1990** [comment]
@gsmet ping :-)

File: integration-tests/main/src/main/java/io/quarkus/it/rest/RestInterface.java [hunk 3]
```diff
@@ -18,6 +19,19 @@
     @GET
     String get();
 
+    @GET
+    CompletionStage<String> asyncGet();
+
+    @GET
+    @Path("/jackson")
+    @Produces("application/json")
+    TestResource.MyData getData();
```
Comment (kenfinnigan) [inline integration-tests/main/src/main/java/io/quarkus/it/rest/RestInterface.java]: Is `getData()` called by a test?

Comment (machi1990) [reply integration-tests/main/src/main/java/io/quarkus/it/rest/RestInterface.java]: Hi @kenfinnigan thanks for the review. This is not used anywhere. Good catch. It has been removed now.

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**